tests/python/unittest/test_contrib_svrg_module.py

```diff
@@ -0,0 +1,295 @@
+    weights = np.array([1.0, 2.0])
+    train_label = train_data.dot(weights)
+
+    di = mx.io.NDArrayIter(train_data, train_label, batch_size=32, shuffle=True, label_name='lin_reg_label')
+    X = mx.sym.Variable('data')
+    Y = mx.symbol.Variable('lin_reg_label')
+    fully_connected_layer = mx.sym.FullyConnected(data=X, name='fc1', num_hidden=1)
+    lro = mx.sym.LinearRegressionOutput(data=fully_connected_layer, label=Y, name="lro")
+
+    mod = SVRGModule(
+        symbol=lro,
+        data_names=['data'],
+        label_names=['lin_reg_label'], update_freq=2)
+    mod.bind(data_shapes=di.provide_data, label_shapes=di.provide_label)
+    mod.init_params(initializer=mx.init.Uniform(0.01), allow_missing=False, force_init=False, allow_extra=False)
+
+    return di, mod
+
+
+def test_bind_module():
+    _, mod = setup()
+    assert mod.binded == True
+    assert mod._mod_aux.binded == True
+
+
+def test_module_init():
+    _, mod = setup()
+    assert mod._mod_aux is not None
+
+
+def test_module_initializer():
+    def regression_model(m):
+        x = mx.symbol.var("data", stype='csr')
+        v = mx.symbol.var("v", shape=(m, 1), init=mx.init.Uniform(scale=.1),
+                          stype='row_sparse')
+        model = mx.symbol.dot(lhs=x, rhs=v)
+        y = mx.symbol.Variable("label")
+        model = mx.symbol.LinearRegressionOutput(data=model, label=y, name="out")
+        return model
+
+    #shape of the data
+    n, m = 128, 100
+    model = regression_model(m)
+
+    data = mx.nd.zeros(shape=(n, m), stype='csr')
+    label = mx.nd.zeros((n, 1))
+    iterator = mx.io.NDArrayIter(data=data, label={'label': label},
+                                 batch_size=n, last_batch_handle='discard')
+
+    # create module
+    mod = SVRGModule(symbol=model, data_names=['data'], label_names=['label'], update_freq=2)
+    mod.bind(data_shapes=iterator.provide_data, label_shapes=iterator.provide_label)
+    mod.init_params()
+    v = mod._arg_params['v']
+    assert v.stype == 'row_sparse'
+    assert np.sum(v.asnumpy()) != 0
+
+
+def test_module_bind():
+    x = mx.sym.Variable("data")
+    net = mx.sym.FullyConnected(x, num_hidden=1)
+
+    mod = SVRGModule(symbol=net, data_names=['data'], label_names=None, update_freq=2)
+    assertRaises(TypeError, mod.bind, data_shapes=['data', mx.nd.zeros(shape=(2, 1))])
+
+    mod.bind(data_shapes=[('data', (2, 1))])
+    assert mod.binded == True
+    assert mod._mod_aux.binded == True
+
+@with_seed()
+def test_module_save_load():
+    x = mx.sym.Variable("data")
+    y = mx.sym.Variable("softmax_label")
+    net = mx.sym.FullyConnected(x, y, num_hidden=1)
+
+    mod = SVRGModule(symbol=net, data_names=['data'], label_names=['softmax_label'], update_freq=2)
+    mod.bind(data_shapes=[('data', (1, 1))])
+    mod.init_params()
+    mod.init_optimizer(optimizer='sgd', optimizer_params={'learning_rate': 0.1})
+    mod.update()
+    mod.save_checkpoint('test', 0, save_optimizer_states=True)
```

Comment:
Recommended to use tempfile. This avoid race condition in file access from other tests and also clean up is easy. Something like:
```
import tempfile
import os
....
 tmp = tempfile.mkdtemp()
 tmpfile = os.path.join(tmp, 'svrg_test')
mod.save_checkpoint(tmpfile, .....)
```